### PR TITLE
Add testing for the vLLM integration in Triton CI/CD

### PR DESCRIPTION
**Pull request by pskiran1**

### CODE REVIEW

File: ci/qa_models/vllm_opt/config.pbtxt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+name: "vllm_opt"
+backend: "vllm_triton_backend"
```
Comment (tanmayv25) [inline ci/qa_models/vllm_opt/config.pbtxt:28]: You might just have to provide backend: "vllm" ?

File: ci/L0_backend_vllm/test.sh [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+mkdir -p models/vllm_opt/1/
+cp ../qa_models/vllm_opt/model.json models/vllm_opt/1/
+cp ../qa_models/vllm_opt/config.pbtxt models/vllm_opt
+
+mkdir -p models/add_sub/1/
+cp ../qa_models/add_sub/model.py models/add_sub/1/
+cp ../qa_models/add_sub/config.pbtxt models/add_sub
+
+pip3 install vllm==0.2.0
```
Comment (tanmayv25) [inline ci/L0_backend_vllm/test.sh:48]: You should not need vllm installation in the test script.

File: ci/qa_models/vllm_opt/model.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "model":"facebook/opt-125m",
+    "disable_log_requests": "true",
+    "gpu_memory_utilization": 0.5
```
Comment (tanmayv25) [inline ci/qa_models/vllm_opt/model.json:4]: You can drop "gpu_memory_utilization" from our test model. I saw some failures when using this limit on my machine.

```
100%|################################################################################################################################| 441/441 [00:00<00:00, 1.10MB/s]
Downloading pytorch_model.bin: 100%|#####################################################################################################################################| 251M/251M [00:02<00:00, 114MB/s]
INFO 10-08 07:11:33 llm_engine.py:205] # GPU blocks: 19794, # CPU blocks: 7281
E1008 07:11:39.335147 94 backend_model.cc:634] ERROR: Failed to create instance: Stub process 'vllm_opt_0_0' is not healthy.
E1008 07:11:39.345423 94 model_lifecycle.cc:621] failed to load 'vllm_opt' version 1: Internal: Stub process 'vllm_opt_0_0' is not healthy.
```

**Comment by tanmayv25** [comment]
After addressing the comments and getting a successful CI run, these changes looks good to be merged for r23.10.
We should add more extensive testing for the backend in our next iteration:
1. Test request cancellation.
2. Test the message when model.json is malformed.
3. Test model load/unload stress test. 
4. When conda gets resolved, then test the backend composibility against other backends.
5. Add performance and accuracy test on the backend on some popular models to compare Triton's impact.

**Review by tanmayv25 — APPROVED** [review]

File: ci/qa_models/vllm_opt/config.pbtxt (hunk 1 above)
Comment (pskiran1) [reply ci/qa_models/vllm_opt/config.pbtxt]: Updated 👍

File: ci/L0_backend_vllm/test.sh (hunk 2 above)
Comment (pskiran1) [reply ci/L0_backend_vllm/test.sh]: Updated 👍

File: ci/qa_models/vllm_opt/model.json (hunk 3 above)
Comment (pskiran1) [reply ci/qa_models/vllm_opt/model.json]: Updated 👍

**Comment by pskiran1** [comment]
The CI pipelines have been executed successfully. Proceeding to merge the changes.

**Pull request merged**